mypy/build.py

```diff
@@ -861,19 +863,68 @@ def is_file(path: str) -> bool:
     return res
 
 
-def find_module(id: str, lib_path_arg: Iterable[str]) -> Optional[str]:
+USER_SITE_PACKAGES = \
+    'from __future__ import print_function; import site; print(site.getusersitepackages());' \
+    'print(*site.getsitepackages(), sep="\\n")'
+VIRTUALENV_SITE_PACKAGES = \
+    'from distutils.sysconfig import get_python_lib; print(get_python_lib())'
+
+
+def call_python(python_executable: str, command: str) -> str:
+    return subprocess.check_output([python_executable, '-c', command],
+                                   stderr=subprocess.PIPE).decode()
+
+
+@functools.lru_cache(maxsize=None)
+def get_package_dirs(python_executable: str) -> List[str]:
+    """Find package directories for given python
+
+    This defaults to the Python running mypy.
+    """
+    if python_executable == sys.executable:
+        # Use running Python's package dirs
+        try:
+            user_dir = site.getusersitepackages()
+            return site.getsitepackages() + [user_dir]
+        except AttributeError:
```

Comment:
`site` won't have either `getusersitepackages` nor `getsitepackages` if it is in a virtualenv.

Reply:
Can you add a comment? This is a bit obscure. Also if either of those exist but the call raises an unrelated AttributeError this would also hide that (though that's low probability).